### DISCUSSION TITLE
No already StaMPS PS processing detected

### DISCUSSION THREAD

**Discussion by AllendePrieto** [alexisInSAR/EZ-InSAR#49] [Help]
Hi Alex!
in the "PS approach/Preparation" step, it gives me the following error:
"No already StaMPS PS processing detected”
and in the matlab environment:
![image](https://github.com/alexisInSAR/EZ-InSAR/assets/131383377/58e11724-5840-4803-ad8c-b99341c79f8f)
Could it be related to the gcc version?

**Comment by alexisInSAR** [comment]
Hi, 

Not exactly. This is related to the version of libstdc++. There is a version conflict between your Linux library and the library provided by MATLAB. My advice is to create a temporary symbolic link between the MATLAB library /usr/local/MATLAB/R2022b/sys/os/glnxa64/libstdc++.s.6 and the same library in your Linux. 

Alexis

**Comment by AllendePrieto** [comment]
Thank you! I have modified it and it still working, but the message is the same when I get to the Run key:

![image](https://github.com/alexisInSAR/EZ-InSAR/assets/131383377/928752b0-b289-4d76-8ffc-8d8d3766f824)
![image](https://github.com/alexisInSAR/EZ-InSAR/assets/131383377/90ec9f14-22f3-4689-9541-f405039be3f9)
![image](https://github.com/alexisInSAR/EZ-InSAR/assets/131383377/e08768e0-c8fd-4428-821e-8b44ff7e6ea4)

What can happen?

**Comment by alexisInSAR** [comment]
Hi, 

What are the sizes of these files? I suspect that your PATCH 1 does not contain PS points. 

Kind regards, 

Alexis

**Comment by AllendePrieto** [comment]
These are the files and their size

![image](https://github.com/alexisInSAR/EZ-InSAR/assets/131383377/7d07edda-edea-48dd-b9f3-9792933ad22d)

**Comment by alexisInSAR** [comment]
This path does not contain PS points. Please modify the number of paths to avoid an empty patch. 

Alexis

**Reply by AllendePrieto** [reply]
Hi Alex.
I can't generate the PS points even though I change the number of paths (7x3) (8x4) ...etc...

What can happen?

![image](https://github.com/alexisInSAR/EZ-InSAR/assets/131383377/dbb53e8b-2e71-4c8b-9011-1ce229857081)

**Reply by alexisInSAR** [reply]
Hi, 

Sorry for my late reply. Can you try just one patch?

Kind regards, 

Alexis

**Reply by AllendePrieto** [reply]
Hi again!
It gives me the same result. It does not generate PS points and the message is the same "No already StaMPS PS processing detected".  My gcc version is 11.3.0.  I tried to change to the gcc-7 version and try again, but it returns the error: "E: Package 'gcc-7' has no installation candidate".  
The solution proposed in the installation instructions link does not work for me at the moment.  Do you know any other solution for this problem?

Regards

**Reply by alexisInSAR** [reply]
Hi, 

StaMPS must be compiled with the version 7 of gcc/g++. What are your Linux version? 

Kind regards, 

Alexis

**Reply by AllendePrieto** [reply]
Hi!
Ubuntu 22.04.2 LTS
is the version

**Comment by quadriel** [comment]
Hi Alex.

I'm getting the same error too. I am using Ubuntu 20.04 and gcc 7.0. About 2 years ago, when I got a similar error while using StaMPS, I solved the problem using python version 2.7. Do you think it would work to build InSARenv in 2.7 environment?

Best regards.

**Reply by AllendePrieto** [reply]
Fixed!
I uninstalled the version I had and installed the gcc-7 and g++-7 version by following the instructions in
https://askubuntu.com/questions/1406962/install-gcc7-on-ubuntu-22-04

Thanks!!

**Reply by quadriel** [reply]
Nice. Thanks for sharing your solution.
I will try right away
